fix(SearchTrack): youtube urls

Xorsiphus opened the pull request:
simple solution for https://github.com/Clytage/rawon/issues/1113
works for me)

Xorsiphus committed:
restore youtube links

mzrtamp commented on src/utils/handlers/general/searchTrack.ts:
```diff
@@ -17,7 +17,7 @@ export async function searchTrack(
     const queryData = checkQuery(query);
     if (queryData.isURL) {
-        const url = new URL(query);
+        let url = new URL(query);
```
```suggestion
        const url = new URL(query);
```
Use const instead

Mednoob replied:
@mzrtamp the use of `let` in his solution is correct since he changed the value of `url` variable on line 81.

Mednoob replied:
but, the declaration of `url` can be changed to `const` after my change request is done.

Mednoob commented on src/utils/handlers/general/searchTrack.ts:
```diff
@@ -77,6 +77,9 @@ export async function searchTrack(
+                        }
                         const track = await youtube.getVideo(
                             /youtu\.be/g.exec(url.hostname) ? url.pathname.replace("/", "") : url.toString()
                         );
```
Instead of changing the `url` variable, you can just do this in the ternary:
```suggestion
                        const track = await youtube.getVideo(
                            /youtu\.be/g.exec(url.hostname) ? url.pathname.replace("/", "") : url.searchParams.get("v")
                        );
```

Xorsiphus committed:
Update src/utils/handlers/general/searchTrack.ts

Well, this way is even better than mine, but eslint swears at types, so you can add the Null-coalescing operator after the third operand

Co-authored-by: Ahmad Yaqdhan <ahmadyaqdhan123@gmail.com>

Xorsiphus committed:
fix compilation error

Xorsiphus reviewed (commented):
😶

Xorsiphus committed:
declared url as const

mzrtamp reviewed (approved):
I'm done with it
